File: ShadowFrameworkJS/src/shadow/system/data/objects/SFVertex3fData.js
//Java to JS on 15/03/2012

function SFVertex3fData(){
	this.vertex=new SFVertex3f();
	this.floatValues=this.vertex.v;
}

inherit(SFVertex3fData,SFVectorData);


SFVertex3fData.prototype["getVertex3f"]=function(){
			return this.vertex;
		};
		
SFVertex3fData.prototype["clone"]=function(){
			return new SFVertex3fData();
		};

File: trunk/SFDataApplication/src/shadow/system/data/objects/SFVertex3fData.java
package shadow.system.data.objects;

import shadow.math.SFVertex3f;

public class SFVertex3fData extends SFVectorData{
	
	private SFVertex3f vertex;
	
	public SFVertex3fData(float x,float y,float z) {
		super(0);
		vertex=new SFVertex3f();
		floatValues=vertex.getV();
		vertex.set3f(x,y,z);
	}
	
	public SFVertex3fData() {
		super(0);
		vertex=new SFVertex3f();
		floatValues=vertex.getV();
	}

	public SFVertex3f getVertex3f(){
		return vertex;
	}
	
	@Override
	public SFVectorData copyDataObject() {
		return new SFVertex3fData();
	}
}
